### PR TITLE
Gateway UT fixes

### PR DESCRIPTION
**Pull request by akshayhavile**
root@akshay-dev-box:/var/work/load-balancer-and-ingress-services-for-kubernetes# make gatewayapitests
sudo docker run \
-w=/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes \
-v /var/work/load-balancer-and-ingress-services-for-kubernetes:/go/src/github.com/vmware/load-balancer-and-ingress-services-for-kubernetes golang:bullseye \
go test -mod=vendor github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/... -failfast -timeout 0 -coverprofile cover-20.out -coverpkg=./...
	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests		coverage: 0.0% of statements
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/graphlayer	244.725s	coverage: 17.5% of statements in ./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/ingestion	270.453s	coverage: 3.0% of statements in ./...
ok  	github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/status	310.656s	coverage: 12.0% of statements in ./...

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id check

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by akshayhavile** [comment]
build ako

File: tests/gatewayapitests/utils.go [hunk 1]
```diff
@@ -126,19 +126,21 @@ func UnsetListenerHostname(l *gatewayv1.Listener) {
...
-		}
 		listener := gatewayv1.Listener{
 			Name:     gatewayv1.SectionName(fmt.Sprintf("listener-%d", port)),
 			Port:     gatewayv1.PortNumber(port),
 			Protocol: gatewayv1.ProtocolType("HTTPS"),
-			Hostname: (*gatewayv1.Hostname)(&hostname),
 		}
+		if !samehost {
+			hostname := fmt.Sprintf("foo-%d.com", port)
+			listener.Hostname = (*gatewayv1.Hostname)(&hostname)
```
Comment (pkoshtavmware) [inline tests/gatewayapitests/utils.go:138]: this line can be taken out from if:else ladder

Comment (pkoshtavmware) [reply tests/gatewayapitests/utils.go:138]: Could you please explain how is this change fixing the ut

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

File: tests/gatewayapitests/utils.go (hunk 1 above)
Comment (akshayhavile) [reply tests/gatewayapitests/utils.go:138]: @pkoshtavmware we are using address of the variable. so previously, all hostname entries in multiple listener scenario was getting overwritten with last hostname in the list. 
So now for each entry, we are allocating new variable and assigning it. Scope of that variable is limited to if loop. So added that logic inside if and else.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Comment by pkoshtavmware** [comment]
build ako

**Comment by akshayhavile** [comment]
failed in`TestCreateUpdateDeleteHostRuleForEvh`. Not related with my changes.. so retriggering

**Comment by akshayhavile** [comment]
build ako

**Review by DixitAakash — APPROVED** [review]
LGTM

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Pull request merged**